languages/python/oso/tests/sugar.polar

```diff
@@ -0,0 +1,67 @@
+allow(actor, action, resource) if
+  permission(actor, action, resource);
+
+role(actor, role, resource) if
+  actor.has_role_for_resource(name: role, resource: resource);
+
+role(user: User, role, resource) if
+  team in user.teams and
+  role(team, role, resource);
+
+Org {
+  roles = ["owner", "member"];
+  permissions = ["invite", "create_repo"];
+  relations = { owns: User };
+
+  "member" if "owner";
+
+  "invite" if "owner";
+  "create_repo" if "member";
+}
+
+relation(user: User, "owns", org: Org) if
+  user = org.owner;
+
+Repo {
```

Comment:
Groups are already here: https://github.com/osohq/oso/blob/d9e252fd02103a92871a4eb113117db2304c5458/languages/python/oso/tests/sugar.polar#L7-L9

Did you have something else in mind?

Reply:
Oh, nice! I ctrl+f-ed for group and failed.

I'm wondering whether `team in user.teams` should be extracted into it's own relation (with rule prototypes), so that you end up with:

```
role(user: User, role, resource) if relation(user, "member", team) and role(role, role, resource);
```


I'm not sure if we have any sugar cases that would use that relation though. So your implementation might be our best practice.

Reply:
TODO: We should have a doc that explains the N rules you need to add to represent groups. E.g., `has_role/3` for groups & adding another `has_role/3` for users that says the user has a role if the user is in a group that has the role.

Example of difficulty in representing user-group relationships via sugar syntax:
```ruby
User {
  relations = { member_of_team: Team };
}

has_relation(user: User, "member_of_team", team: Team) if
  team in user.teams;

Team {
  relations = { repo: Repo };
}

Repo {
  # ???
}
```

Reply:
Knock-on implication for #1084 

- Do we ever need to register groups as a 'Group' entity if we only ever deal with groups in rule bodies? I.e., we never specialize on a `Group` type.

Reply:
Took an note to follow up on this